File: other/Solution7.java
public class Solution7{
	 public int reverse(int x) {
	 	if(x == 0){
	 		return x;
	 	}
	 	int copy = x;
	 	long result = 0;
	 	while(copy != 0){
	 		result = result * 10;
	 		int digit = copy % 10;
	 		result += digit;
	 		copy = copy / 10;
	 	}
	 	if((int) result == result){
             return (int) result;
         }
         else{
             return 0;
         }
    }
}

File: C++/Solution7.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution7 {
public:
    int reverse(int x) {
        if(x == INT_MAX || x == INT_MIN){
            return 0;
        }
        if(x < 0){
            return -reverse(-x);
        }
        long result = 0;
        while(x != 0){
            result  = result * 10 + x % 10;
            x = x / 10;
        }
        if(result >= INT_MAX){
            return 0;
        }
        return int(result);
    }
};

int main(){
    Solution7 s;
    cout << s.reverse(123);
}